### PR TITLE
Add e8 profile for ocp4

### PR DESCRIPTION
**Pull request by jaedolph**
#### Description:

- Porting 'Australian Cyber Security Centre (ACSC) Essential Eight' profile to ocp4

#### Rationale:

- Allow e8 profile to be used to scan RHCOS nodes on OCP4

- Also added the ocp platform to some OVAL checks that were not getting built for this profile.

### CODE REVIEW

File: ocp4/profiles/e8.profile [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    that align to the Australian Cyber Security Centre (ACSC) Essential Eight.
+
+    A copy of the Essential Eight in Linux Environments guide can be found at the
+    ACSC website:
+
+    https://www.cyber.gov.au/publications/essential-eight-in-linux-environments
+
+selections:
+
+  ### Remove obsolete packages
```
Comment (jaedolph) [inline ocp4/profiles/e8.profile:16]: These are N/A for RHCOS

File: ocp4/profiles/e8.profile [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  #- ensure_redhat_gpgkey_installed
+  #- ensure_gpgcheck_never_disabled
+  #- ensure_gpgcheck_local_packages
+  #- ensure_gpgcheck_globally_activated
+  #- security_patches_up_to_date
+  #- dnf-automatic_security_updates_only
+
+  ### System security settings
+  - sysctl_kernel_randomize_va_space
+#  - sysctl_kernel_exec_shield
```
Comment (jaedolph) [inline ocp4/profiles/e8.profile:43]: This check is not available for OCP4 platform. Also I believe it is not possible to use sysctl to turn off exec shield in RHCOS and RHEL8 and it defaults to enabled

File: ocp4/profiles/e8.profile [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+  - var_selinux_state=enforcing
+  - selinux_state
+  - var_selinux_policy_name=targeted
+  - selinux_policytype
+
+  ### Filesystem integrity
+  #- rpm_verify_hashes
+  #- rpm_verify_permissions
+  #- rpm_verify_ownership
+  #- file_permissions_unauthorized_sgid
```
Comment (jaedolph) [inline ocp4/profiles/e8.profile:61]: No checks for these on OCP4 platform. May be worth adding ocp platform to the rule.yaml definitions for them

File: ocp4/profiles/e8.profile [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+  #- mount_option_dev_shm_noexec
+
+  ### Network
+  #- package_firewalld_installed
+  #- service_firewalld_enabled
+  #- network_sniffer_disabled
+
+  ### Admin privileges
+  - accounts_no_uid_except_zero
+  #- sudo_remove_nopasswd
```
Comment (jaedolph) [inline ocp4/profiles/e8.profile:85]: RHCOS uses sshkeys by default so can't supply password for sudo

File: ocp4/profiles/e8.profile [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+  - sshd_disable_empty_passwords
+  - sshd_disable_user_known_hosts
+  - sshd_enable_strictmodes
+
+  # See also: https://www.cyber.gov.au/ism/guidelines-using-cryptography
+  - var_system_crypto_policy=future
+  - configure_crypto_policy
+  - configure_ssh_crypto_policy
+
+  ### Application whitelisting
```
Comment (jaedolph) [inline ocp4/profiles/e8.profile:139]: No fapolicyd on RHCOS

File: ocp4/profiles/e8.profile [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+  - sshd_use_priv_separation
+  - sshd_do_not_permit_user_env
+  - sshd_disable_rhosts
+  - sshd_set_loglevel_info
+  - sshd_disable_empty_passwords
+  - sshd_disable_user_known_hosts
+  - sshd_enable_strictmodes
+
+  # See also: https://www.cyber.gov.au/ism/guidelines-using-cryptography
+  - var_system_crypto_policy=future
```
Comment (jaedolph) [inline ocp4/profiles/e8.profile:135]: DEFAULT:NO-SHA1 is not supported on RHCOS yet. 'FUTURE' uses crypto compliant with the ISM (SHA1 disabled)

File: ocp4/profiles/e8.profile (hunk 2 above)
Comment (JAORMX) [reply ocp4/profiles/e8.profile:43]: you can do sysctl setup with OCP4. There is an operator for that.

**Comment by jaedolph** [comment]
@JAORMX not sure why the tests are failing. Do you think this was an issue with the CI/CD or something else?

File: ocp4/profiles/e8.profile (hunk 2 above)
Comment (jaedolph) [reply ocp4/profiles/e8.profile:43]: My understanding is Red Hat has disabled the ability to turn exec shield off from RHEL7 onwards, so there is no need to check it is turned on: https://access.redhat.com/solutions/974563

Comment (jaedolph) [reply ocp4/profiles/e8.profile:43]: what i meant by 

> This check is not available for OCP4 platform.

is that ocp4 is not included as a platform for this rule:
https://github.com/ComplianceAsCode/content/blob/master/linux_os/guide/system/permissions/restrictions/enable_execshield_settings/sysctl_kernel_exec_shield/rule.yml

**Comment by shawndwells** [comment]
@openscap-ci retest this please

**Comment by shawndwells** [comment]
@shaneboulden You have the perspective/context on what should go into an E8 profile. Long as this builds, and you ack the rule selections, I'm good merging this PR.

**Comment by shaneboulden** [comment]
LGTM

We'd love to have fapolicyd/whitelisting available in RHCOS, and I like that it's commented here instead of being removed.

**Comment by yuumasato** [comment]
Per approvals I'm merging.

**Pull request merged**